dapps/marketplace/src/pages/nav/Nav.js

```diff
@@ -66,12 +67,67 @@ const Nav = ({
   history
 }) => {
   const [open, setOpen] = useState()
+  const [consoleOpen, setConsoleOpen] = useState(false)
+  const [consoleLogConnected, setConsoleLogConnected] = useState(false)
+
+  const screenConsoleEnabled = localStorage.screenConsole
+
+  useEffect(() => {
+    if (wallet && !consoleLogConnected && screenConsoleEnabled) {
+      ConsoleLogCatcher().connect((method, logString) => {
+        const logs = JSON.parse(localStorage.getItem('capturedLogs') || '[]')
+        // only keep max 15 items in the logs
+        logs.slice(Math.max(0, logs.length - 15))
```

Comment:
nit: slice accepts a negative index. So you can just use logs.slice(-15) and it will only keep the last 15 elements.

Reply:
that is very useful thanks!